colcon_sanitizer_reports/_sanitizer_section_part_stack_trace.py

```diff
@@ -66,3 +70,4 @@ def __init__(self, lines: Tuple[str, ...]) -> None:
         assert key is not None, 'Could not find key in given stack trace lines.'
 
         self.key = key
+        self.lines = lines
```

Comment:
It's better to avoid public attributes whenever possible. Protected attribute + getter is better as it prevents having other classes modifying the attribute value.

E.g.

```
~ ❯❯❯ cat foo.py                                                              ⏎
class A:
  def __init__(self):
    self._x = 42

  @property
  def x(self):
    return self._x

a = A()
print(a.x)
a.x = 'abcd'
~ ❯❯❯ python3 foo.py
42
Traceback (most recent call last):
  File "foo.py", line 11, in <module>
    a.x = 'abcd'
AttributeError: can't set attribute
```

Reply:
Agreed. I recently got into this habit with python 3.7 frozen dataclasses. If I fix it here, I'd like to fix it everywhere (~4 occurrences in this codebase). Mind if I do it in this PR?

Reply:
SGTM